File: app/src/main/java/com/sc/clgg/activity/usersettings/ModifyPasswordActivity.java
//package com.sc.clgg.activity.usersettings;
//
//import android.content.Intent;
//import android.text.Editable;
//import android.text.TextUtils;
//import android.text.TextWatcher;
//import android.view.View;
//import android.widget.EditText;
//import android.widget.ImageButton;
//
//import com.alibaba.fastjson.JSON;
//import com.sc.clgg.R;
//import com.sc.clgg.activity.basic.LoginActivity;
//import com.sc.clgg.base.BaseActivity;
//import com.sc.clgg.dialog.LoadingDialogHelper;
//import com.sc.clgg.http.HttpCallBack;
//import com.sc.clgg.http.HttpRequestHelper;
//import com.sc.clgg.util.ConfigUtil;
//import com.sc.clgg.util.Tools;
//
//import java.util.Map;
//
//import butterknife.BindView;
//import butterknife.OnClick;
//
//public class ModifyPasswordActivity extends BaseActivity {
//    @BindView(R.id.et_old_password) EditText mEtOldPassword;
//    @BindView(R.id.et_new_password) EditText mEtNewPassword;
//    @BindView(R.id.et_again_new_password) EditText mEtAgainNewPassword;
//    @BindView(R.id.iv_delete_old_password) ImageButton mIvDeleteOldPassword;
//    @BindView(R.id.iv_delete_new_password) ImageButton mIvDeleteNewPassword;
//    @BindView(R.id.iv_delete_again_new_password) ImageButton mIvDeleteAgainNewPassword;
//    private LoadingDialogHelper mLoadingDialogHelper;
//
//    @Override
//    public void initTitle() {
//        setTitle(getString(R.string.modify_password));
//    }
//
//    @Override
//    protected int layoutRes() {
//        return R.layout.activity_modify_password;
//    }
//
//    @Override
//    protected void init() {
//        mLoadingDialogHelper = new LoadingDialogHelper(this);
//        initListener();
//    }
//
//    @OnClick({R.id.iv_delete_old_password, R.id.iv_delete_new_password, R.id.iv_delete_again_new_password})
//    void b(View v) {
//        switch (v.getId()) {
//            case R.id.iv_delete_old_password:
//                mEtOldPassword.setText("");
//                break;
//
//            case R.id.iv_delete_new_password:
//                mEtNewPassword.setText("");
//                break;
//
//            case R.id.iv_delete_again_new_password:
//                mEtAgainNewPassword.setText("");
//                break;
//
//            default:
//                break;
//        }
//    }
//
//    private void initListener() {
//        mEtOldPassword.addTextChangedListener(new TextWatcher() {
//            @Override
//            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
//
//            }
//
//            @Override
//            public void onTextChanged(CharSequence s, int start, int before, int count) {
//
//            }
//
//            @Override
//            public void afterTextChanged(Editable s) {
//                if (s.length() > 0) {
//                    mIvDeleteOldPassword.setVisibility(View.VISIBLE);
//                } else {
//                    mIvDeleteOldPassword.setVisibility(View.INVISIBLE);
//                }
//            }
//        });
//        mEtNewPassword.addTextChangedListener(new TextWatcher() {
//            @Override
//            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
//
//            }
//
//            @Override
//            public void onTextChanged(CharSequence s, int start, int before, int count) {
//
//            }
//
//            @Override
//            public void afterTextChanged(Editable s) {
//                if (s.length() > 0) {
//                    mIvDeleteNewPassword.setVisibility(View.VISIBLE);
//                } else {
//                    mIvDeleteNewPassword.setVisibility(View.INVISIBLE);
//                }
//            }
//        });
//        mEtAgainNewPassword.addTextChangedListener(new TextWatcher() {
//            @Override
//            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
//
//            }
//
//            @Override
//            public void onTextChanged(CharSequence s, int start, int before, int count) {
//
//            }
//
//            @Override
//            public void afterTextChanged(Editable s) {
//                if (s.length() > 0) {
//                    mIvDeleteAgainNewPassword.setVisibility(View.VISIBLE);
//                } else {
//                    mIvDeleteAgainNewPassword.setVisibility(View.INVISIBLE);
//                }
//            }
//        });
//    }
//
//    @OnClick(R.id.confirm_btn)
//    void a() {
//        resetPassword();
//    }
//
//    public void resetPassword() {
//        String newPassword = mEtNewPassword.getText().toString();
//        String oldPassword = mEtOldPassword.getText().toString();
//        String againNewPassword = mEtAgainNewPassword.getText().toString();
//        if (TextUtils.isEmpty(oldPassword)) {
//            Tools.Toast("原始密码不能为空");
//            return;
//        }
//        if (newPassword.length() < 6) {
//            Tools.Toast("密码长度应在6-16位，当前" + newPassword.length() + "位！");
//            return;
//        }
//        if (TextUtils.isEmpty(newPassword)) {
//            Tools.Toast("新密码不能为空");
//            return;
//        }
//        if (TextUtils.isEmpty(againNewPassword)) {
//            Tools.Toast("第二次输入新密码不能为空");
//            return;
//        }
//        if (!newPassword.equals(againNewPassword)) {
//            Tools.Toast("两次密码不一致");
//            return;
//        }
//
//        if (newPassword.equals(oldPassword)) {
//            Tools.Toast("新旧密码不可相同");
//            return;
//        }
//
//        HttpRequestHelper.modifyPassword(new ConfigUtil().getUsername(), oldPassword, newPassword, new HttpCallBack() {
//            @Override
//            public void onStart() {
//                super.onStart();
//                mLoadingDialogHelper.show();
//            }
//
//            @Override
//            public void onFinish() {
//                super.onFinish();
//                super.onStart();
//                mLoadingDialogHelper.dismiss();
//            }
//
//            @Override
//            public void onSuccess(String body) {
//                Map<String, Object> map = (Map<String, Object>) JSON.parse(body);
//                if ((boolean) map.get("status")) {
//                    Tools.Toast("密码修改成功");
//                    startActivity(new Intent(ModifyPasswordActivity.this, LoginActivity.class));
//                    finish();
//                } else {
//                    Tools.Toast(String.valueOf(map.get("msg")));
//                }
//            }
//        });
//    }
//}

File: app/src/main/java/com/sc/clgg/activity/usersettings/ModifyPasswordActivity.kt
package com.sc.clgg.activity.usersettings

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.google.gson.Gson
import com.sc.clgg.R
import com.sc.clgg.activity.basic.LoginActivity
import com.sc.clgg.bean.StatusBean
import com.sc.clgg.dialog.LoadingDialogHelper
import com.sc.clgg.http.HttpCallBack
import com.sc.clgg.http.HttpRequestHelper
import com.sc.clgg.util.ConfigUtil
import com.sc.clgg.util.Tools
import com.sc.clgg.util.setTextChangeListener
import kotlinx.android.synthetic.main.activity_modify_password.*
import org.jetbrains.anko.sdk25.coroutines.onClick
import org.jetbrains.anko.toast

class ModifyPasswordActivity : AppCompatActivity() {
    private var mLoadingDialogHelper: LoadingDialogHelper? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        setContentView(R.layout.activity_modify_password)
        title = getString(R.string.modify_password)
        super.onCreate(savedInstanceState)

        mLoadingDialogHelper = LoadingDialogHelper(this)
        initListener()
    }

    private fun initListener() {
        et_old_password!!.setTextChangeListener {
            iv_delete_old_password!!.visibility = if (it.isNotEmpty()) View.VISIBLE else View.INVISIBLE
        }

        et_new_password.setTextChangeListener {
            iv_delete_new_password.visibility = if (it.isNotEmpty()) View.VISIBLE else View.INVISIBLE
        }

        et_again_new_password.setTextChangeListener {
            iv_delete_again_new_password!!.visibility = if (it.isNotEmpty()) View.VISIBLE else View.INVISIBLE
        }

        iv_delete_old_password.onClick { et_old_password!!.setText("") }
        iv_delete_new_password.onClick { et_new_password!!.setText("") }
        iv_delete_again_new_password.onClick { et_again_new_password!!.setText("") }

        confirm_btn.onClick {
            determine()
        }
    }

    private fun determine() {
        val oldPassword = et_old_password!!.text.toString()
        val newPassword = et_new_password!!.text.toString()
        val againNewPassword = et_again_new_password!!.text.toString()

        if (oldPassword.isEmpty()) {
            Tools.Toast("原始密码不能为空")
            return
        }
        if (newPassword == oldPassword) {
            Tools.Toast("新旧密码不可相同")
            return
        }
        if (newPassword.isEmpty()) {
            Tools.Toast("新密码不能为空")
            return
        }
        if (newPassword.length < 6) {
            Tools.Toast("密码长度应在6-16位，当前" + newPassword.length + "位！")
            return
        }
        if (againNewPassword.isEmpty()) {
            Tools.Toast("第二次输入新密码不能为空")
            return
        }
        if (againNewPassword != newPassword) {
            Tools.Toast("两次密码不一致")
            return
        }

        HttpRequestHelper.modifyPassword(ConfigUtil().username, oldPassword, newPassword, object : HttpCallBack() {
            override fun onStart() {
                super.onStart()
                mLoadingDialogHelper!!.show()
            }

            override fun onFinish() {
                super.onFinish()
                super.onStart()
                mLoadingDialogHelper!!.dismiss()
            }

            override fun onSuccess(body: String) {
                val statusBean = Gson().fromJson(body, StatusBean::class.java)
                if (statusBean.status) {
                    toast("密码修改成功")
                    startActivity(Intent(this@ModifyPasswordActivity, LoginActivity::class.java))
                    finish()
                } else {
                    toast(statusBean.msg!!)
                }
            }
        })
    }
}
